api/src/services/fetchText.ts

```diff
@@ -1,34 +1,22 @@
-import { Request, Response } from 'express'
-import { fetchEntryText } from './fetchEntry'
-import { Knex } from 'knex'
+import graphRepository, { Entry } from './GraphRepository'
 
-export default async (req:Request, res:Response, pg:Knex, timeLogID:string) => {
-  let entry = await fetchEntryText(req.params.name, pg)
-    .catch(err => {throw err})
-  let newEntry = false
+export default async (name: string): Promise<Entry> => {
+    let entry = await graphRepository.findByName(name)
 
-  if (entry.name === '') {
-    const addEntry = async () => {
-      await pg('graphs').insert({
-        'text': '',
-        'img': '',
-        'name': req.params.name,
-        'outputUpdated': false
-      }).then((item:object) => {
-        console.timeLog(timeLogID)
```

Comment:
Da się specjalnie owrapować całą klasę, by Ci automatycznie liczyło ile czasu zajmuje jakaś funckcja.
Mogę Cię nauczyć jak to zrobić.
Bo to co tu się dzieje, to na razie strasznie mgli kod, a nie wiem czy jest tak bardzo potrzebne.

Reply:
Takie rzeczy fajnie się robi:
https://www.typescriptlang.org/docs/handbook/decorators.html

czy

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy

czy jak masz dobre granularne funkcje to po prostu

https://www.codecademy.com/learn/game-dev-learn-javascript-higher-order-functions-and-iterators/modules/game-dev-learn-javascript-iterators/cheatsheet#:~:text=In%20Javascript%2C%20functions%20can%20be,and%2For%20returns%20a%20function.

Ale pokażę to na przykładzie